File: benchmark/python/polynomial_eval.py
import time
coeffs = [i + 1 for i in range(1001)]
def horner(coeffs, x):
    result = coeffs[-1]
    for i in range(len(coeffs) - 2, -1, -1):
        result = result * x + coeffs[i]
    return result
start = time.time() * 1000
s = 0
for _ in range(100000):
    s += horner(coeffs, 1.5)
end = time.time() * 1000
print(f"Sum: {s}")
print(f"Time: {end - start:.0f}ms")

File: benchmark/php/polynomial_eval.php
<?php
$coeffs = [];
for ($i = 0; $i <= 1000; $i++) { $coeffs[] = $i + 1; }
function horner($coeffs, $x) {
    $result = $coeffs[1000];
    for ($i = 999; $i >= 0; $i--) {
        $result = $result * $x + $coeffs[$i];
    }
    return $result;
}
$start = hrtime(true);
$sum = 0;
for ($k = 0; $k < 100000; $k++) {
    $sum += horner($coeffs, 1.5);
}
$end = hrtime(true);
$ms = ($end - $start) / 1e6;
echo "Sum: " . $sum . "
";
echo "Time: " . round($ms) . "ms
";
